tests/framework/TestDiffer/TestXMLDiffer.py

```diff
@@ -19,91 +19,97 @@
+  def testCompareOrderedElement(self):
+    tree1 = ET.fromstring("<test></test>")
+    same, _ = XMLDiff.compare_ordered_element(tree1, tree1)
+    self.assertTrue(same, "simple")
+
+    hello1 = ET.fromstring("<test>Hello  World</test>")
+    hello2 = ET.fromstring("<test>Hello World</test>")
+    same, _ = XMLDiff.compare_ordered_element(hello1, hello2)
+    self.assertFalse(same, "whitespace")
+    same, _ = XMLDiff.compare_ordered_element(hello1, hello2, remove_whitespace=True)
+    self.assertTrue(same, "whitespace with remove")
+
+  def testCompareUnorderedElement(self):
+    tree1 = ET.fromstring("<test></test>")
+    tree2 = copy.deepcopy(tree1)
+    same, _ = XMLDiff.compare_unordered_element(tree1, tree2)
+    self.assertTrue(same, "simple")
+
+    hello1 = ET.fromstring("<test>Hello  World</test>")
+    hello2 = ET.fromstring("<test>Hello World</test>")
+    same, _ = XMLDiff.compare_unordered_element(hello1, hello2)
+    self.assertFalse(same, "whitespace")
+    same, _ = XMLDiff.compare_unordered_element(hello1, hello2, remove_whitespace=True)
+    self.assertTrue(same, "whitespace with remove")
+
+  def testIgnoreSubnodesFromTree(self):
+    tree1 = ET.fromstring("<test> <a>0</a> <b>0</b> </test>")
+    tree2 = copy.deepcopy(tree1)
+    aTree, bTree, success = XMLDiff.ignore_subnodes_from_tree(tree1, tree2, ignored_nodes=["./b"])
+    self.assertTrue(success, "ignoring child node in both trees")
+    same, _ = XMLDiff.compare_unordered_element(aTree, bTree)
+    self.assertTrue(same, "comparing roots after ignoring child node in both trees")
+
+    treeAttrib1 = ET.fromstring('<test><a n="0">1</a><a n="1"/></test>')
+    treeAttrib2 = ET.fromstring('<test><a n="0">2</a><a n="1"/></test>')
+    aTree, bTree, success = XMLDiff.ignore_subnodes_from_tree(treeAttrib1, treeAttrib2, ignored_nodes=['./a[@n="0"]'])
+    self.assertTrue(success, "ignoring child node in both trees via attribute")
+    same, _ = XMLDiff.compare_unordered_element(aTree, bTree)
+    self.assertTrue(same, "comparing roots after ignoring child node in both trees via attribute")
+
+    treeExclusive1 = ET.fromstring('<test> <a/> <c/> </test>')
+    treeExclusive2 = ET.fromstring('<test> <a/> <b/> </test>')
+    aTree, bTree, success = XMLDiff.ignore_subnodes_from_tree(treeExclusive1, treeExclusive2, ignored_nodes=['./b', './c'])
+    self.assertTrue(success, "ignoring multiple nodes exclusive to trees")
+    same, _ = XMLDiff.compare_unordered_element(aTree, bTree)
+    self.assertTrue(same, "comparing roots after ignoring multiple nodes exclusive to trees")
+
+  def testComparePathsInSubnodes(self):
+    # exactly equal paths
+    path1 = os.path.sep.join(['..', 'some', 'rel', 'path'])
+    path2 = copy.deepcopy(path1)
+    node1 = ET.fromstring(f"<root><test>{path1}</test></root>")
+    node2 = ET.fromstring(f"<root><test>{path2}</test></root>")
+    _, _, equal = XMLDiff.compare_paths_in_subnodes(node1, node2, ["./test"])
```

Comment:
do you think it would be good to add an additional test for when the `nodes` list (the third argument) in the `compare_paths_in_subnodes` call has more than one entry?

Reply:
I think that's a good idea. I'll add a test that includes multiple nodes in that list.